In [1]:
import pandas as pd
import numpy as np

# Checkpoint 1: Data Cleaning 1

### Load the companies and rounds data (provided on the previous page) into two data frames and name them companies and rounds2 respectively.

In [55]:
#reading the companies data set
companies = pd.read_csv("C:/Users/vemul/Desktop/DataScience/Investment Case Study/Companies.txt", 
                        delimiter = '\t',
                       encoding = 'ISO-8859-1')

In [56]:
#reading the rounds2 data set
rounds2 = pd.read_csv("C:/Users/vemul/Desktop/DataScience/Investment Case Study/rounds2.csv",
                     encoding = 'ISO-8859-1')

In [57]:
#checking the column names of companies data set
companies.columns

Index(['permalink', 'name', 'homepage_url', 'category_list', 'status',
       'country_code', 'state_code', 'region', 'city', 'founded_at'],
      dtype='object')

In [58]:
#shape of companies data set
companies.shape

(66368, 10)

In [59]:
companies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 66368 entries, 0 to 66367
Data columns (total 10 columns):
permalink        66368 non-null object
name             66367 non-null object
homepage_url     61310 non-null object
category_list    63220 non-null object
status           66368 non-null object
country_code     59410 non-null object
state_code       57821 non-null object
region           58338 non-null object
city             58340 non-null object
founded_at       51147 non-null object
dtypes: object(10)
memory usage: 5.1+ MB


In [60]:
#checking the companies data set for encoding issues in columns after loading
companies

permalink  \
0                                    /Organization/-Fame   
1                                 /Organization/-Qounter   
2                    /Organization/-The-One-Of-Them-Inc-   
3                                  /Organization/0-6-Com   
4                         /Organization/004-Technologies   
5                       /Organization/01Games-Technology   
6                    /Organization/0Ndine-Biomedical-Inc   
7                                   /Organization/0Xdata   
8                                        /Organization/1   
9                              /Organization/1-2-3-Listo   
10                                 /Organization/1-4-All   
11                        /Organization/1-618-Technology   
12                           /Organization/1-800-Dentist   
13                           /Organization/1-800-Doctors   
14              /Organization/1-800-Publicrelations-Inc-   
15                            /Organization/1-Mainstream   
16                                 /Organization/1-Of-99   
17                             /Organization/10-20-Media   
18                         /Organization/10-Minutes-With   
19                              /Organization/1000-Corks   
20                            /Organization/1000-Markets   
21                                 /Organization/1000Chi   
22                               /Organization/1000Lookz   
23                            /Organization/1000Memories   
24                         /Organization/1000Museums-Com   
25                              /Organization/1001-Menus   
26                      /Organization/1001Pharmacies-Com   
27                                 /Organization/1006-Tv   
28                                /Organization/100Du-Tv   
29                                /Organization/100E-Com   
...                                                  ...   
66338                                /Organization/Zykis   
66339                       /Organization/Zylie-The-Bear   
66340                       /Organization/Zylun-Staffing   
66341                       /Organization/Zyme-Solutions   
66342                             /Organization/Zymergen   
66343                              /Organization/Zymetis   
66344                            /Organization/Zymeworks   
66345                             /Organization/Zymochem   
66346                                /Organization/Zympi   
66347                            /Organization/Zymr-Inc-   
66348                                /Organization/Zyncd   
66349                               /Organization/Zyncro   
66350              /Organization/Zynerba-Pharmaceuticals   
66351            /Organization/Zynergy-Projects-Services   
66352                                /Organization/Zynga   
66353                             /Organization/Zyngenia   
66354                              /Organization/Zynstra   
66355                           /Organization/Zyomyx-Inc   
66356                                 /Organization/Zype   
66357                               /Organization/Zypsee   
66358                     /Organization/Zyraz-Technology   
66359                                /Organization/Zyrra   
66360                               /Organization/Zystor   
66361                           /Organization/Zytoprotec   
66362                                /Organization/Zzish   
66363  /Organization/Zznode-Science-And-Technology-Co...   
66364                          /Organization/Zzzzapp-Com   
66365                               /Organization/ÁEron   
66366                             /Organization/ÔAsys-2   
66367  /Organization/Ä°Novatiff-Reklam-Ve-Tanä±Tä±M-H...   

                                                name  \
0                                              #fame   
1                                           :Qounter   
2                             (THE) ONE of THEM,Inc.   
3                                            0-6.com   
4                                   004 Techno

In [61]:
#resolving the encoding issues in the columns
companies['permalink'] = companies.permalink.str.encode('ISO-8859-1').str.decode('ascii','ignore')
companies['name'] = companies.name.str.encode('ISO-8859-1').str.decode('ascii','ignore')
companies['region'] = companies.region.str.encode('ISO-8859-1').str.decode('ascii','ignore')
companies['city'] = companies.city.str.encode('ISO-8859-1').str.decode('ascii','ignore')

In [62]:
#rechecking for resolution
companies

permalink  \
0                                    /Organization/-Fame   
1                                 /Organization/-Qounter   
2                    /Organization/-The-One-Of-Them-Inc-   
3                                  /Organization/0-6-Com   
4                         /Organization/004-Technologies   
5                       /Organization/01Games-Technology   
6                    /Organization/0Ndine-Biomedical-Inc   
7                                   /Organization/0Xdata   
8                                        /Organization/1   
9                              /Organization/1-2-3-Listo   
10                                 /Organization/1-4-All   
11                        /Organization/1-618-Technology   
12                           /Organization/1-800-Dentist   
13                           /Organization/1-800-Doctors   
14              /Organization/1-800-Publicrelations-Inc-   
15                            /Organization/1-Mainstream   
16                                 /Organization/1-Of-99   
17                             /Organization/10-20-Media   
18                         /Organization/10-Minutes-With   
19                              /Organization/1000-Corks   
20                            /Organization/1000-Markets   
21                                 /Organization/1000Chi   
22                               /Organization/1000Lookz   
23                            /Organization/1000Memories   
24                         /Organization/1000Museums-Com   
25                              /Organization/1001-Menus   
26                      /Organization/1001Pharmacies-Com   
27                                 /Organization/1006-Tv   
28                                /Organization/100Du-Tv   
29                                /Organization/100E-Com   
...                                                  ...   
66338                                /Organization/Zykis   
66339                       /Organization/Zylie-The-Bear   
66340                       /Organization/Zylun-Staffing   
66341                       /Organization/Zyme-Solutions   
66342                             /Organization/Zymergen   
66343                              /Organization/Zymetis   
66344                            /Organization/Zymeworks   
66345                             /Organization/Zymochem   
66346                                /Organization/Zympi   
66347                            /Organization/Zymr-Inc-   
66348                                /Organization/Zyncd   
66349                               /Organization/Zyncro   
66350              /Organization/Zynerba-Pharmaceuticals   
66351            /Organization/Zynergy-Projects-Services   
66352                                /Organization/Zynga   
66353                             /Organization/Zyngenia   
66354                              /Organization/Zynstra   
66355                           /Organization/Zyomyx-Inc   
66356                                 /Organization/Zype   
66357                               /Organization/Zypsee   
66358                     /Organization/Zyraz-Technology   
66359                                /Organization/Zyrra   
66360                               /Organization/Zystor   
66361                           /Organization/Zytoprotec   
66362                                /Organization/Zzish   
66363  /Organization/Zznode-Science-And-Technology-Co...   
66364                          /Organization/Zzzzapp-Com   
66365                                 /Organization/Eron   
66366                               /Organization/Asys-2   
66367  /Organization/Novatiff-Reklam-Ve-TanTM-Hizmetl...   

                                          name  \
0                                        #fame   
1                                     :Qounter   
2                       (THE) ONE of THEM,Inc.   
3                                      0-6.com   
4                             004 Technologies   
5                         

In [63]:
#checking the column names for rounds2 data set
rounds2.columns

Index(['company_permalink', 'funding_round_permalink', 'funding_round_type',
       'funding_round_code', 'funded_at', 'raised_amount_usd'],
      dtype='object')

In [64]:
rounds2.shape

(114949, 6)

In [65]:
rounds2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 114949 entries, 0 to 114948
Data columns (total 6 columns):
company_permalink          114949 non-null object
funding_round_permalink    114949 non-null object
funding_round_type         114949 non-null object
funding_round_code         31140 non-null object
funded_at                  114949 non-null object
raised_amount_usd          94959 non-null float64
dtypes: float64(1), object(5)
memory usage: 5.3+ MB


In [66]:
rounds2

company_permalink  \
0                                     /organization/-fame   
1                                  /ORGANIZATION/-QOUNTER   
2                                  /organization/-qounter   
3                     /ORGANIZATION/-THE-ONE-OF-THEM-INC-   
4                                   /organization/0-6-com   
5                          /ORGANIZATION/004-TECHNOLOGIES   
6                        /organization/01games-technology   
7                     /ORGANIZATION/0NDINE-BIOMEDICAL-INC   
8                     /organization/0ndine-biomedical-inc   
9                                    /ORGANIZATION/0XDATA   
10                                   /organization/0xdata   
11                                   /ORGANIZATION/0XDATA   
12                                   /organization/0xdata   
13                                        /ORGANIZATION/1   
14                                        /organization/1   
15                                        /ORGANIZATION/1   
16                              /organization/1-2-3-listo   
17                                  /ORGANIZATION/1-4-ALL   
18                         /organization/1-618-technology   
19                            /ORGANIZATION/1-800-DENTIST   
20                            /organization/1-800-doctors   
21               /ORGANIZATION/1-800-PUBLICRELATIONS-INC-   
22                             /organization/1-mainstream   
23                                  /ORGANIZATION/1-OF-99   
24                              /organization/10-20-media   
25                              /ORGANIZATION/10-20-MEDIA   
26                              /organization/10-20-media   
27                              /ORGANIZATION/10-20-MEDIA   
28                          /organization/10-minutes-with   
29                          /ORGANIZATION/10-MINUTES-WITH   
...                                                   ...   
114919                              /ORGANIZATION/ZYNSTRA   
114920                           /organization/zyomyx-inc   
114921                           /ORGANIZATION/ZYOMYX-INC   
114922                           /organization/zyomyx-inc   
114923                           /ORGANIZATION/ZYOMYX-INC   
114924                                 /organization/zype   
114925                                 /ORGANIZATION/ZYPE   
114926                               /organization/zypsee   
114927                               /ORGANIZATION/ZYPSEE   
114928                     /organization/zyraz-technology   
114929                     /ORGANIZATION/ZYRAZ-TECHNOLOGY   
114930                     /organization/zyraz-technology   
114931                     /ORGANIZATION/ZYRAZ-TECHNOLOGY   
114932                                /organization/zyrra   
114933                                /ORGANIZATION/ZYRRA   
114934                                /organization/zyrra   
114935                                /ORGANIZATION/ZYRRA   
114936                               /organization/zystor   
114937                           /ORGANIZATION/ZYTOPROTEC   
114938                                /organization/zzish   
114939                                /ORGANIZATION/ZZISH   
114940                                /organization/zzish   
114941  /ORGANIZATION/ZZNODE-SCIENCE-AND-TECHNOLOGY-CO...   
114942                          /organization/zzzzapp-com   
114943                          /ORGANIZATION/ZZZZAPP-COM   
114944                          /organization/zzzzapp-com   
114945                          /ORGANIZATION/ZZZZAPP-COM   
114946                               /organization/ãeron   
114947                             /ORGANIZATION/ÔASYS-2   
114948  /organization/ä°novatiff-reklam-ve-tanä±tä±m-h...   

                                funding_round_permalink   funding_round_type  \
0       /funding-round/9a01d05418af9f794eebff7ace91f638              venture   
1       /funding-round/22dacff496eb7acb2b901dec1dfe5633              venture   
2       /funding-

In [67]:
#resolving encoding issues in columns of rounds2
rounds2['company_permalink'] = rounds2.company_permalink.str.encode('ISO-8859-1').str.decode('ascii','ignore')

In [68]:
rounds2

company_permalink  \
0                                     /organization/-fame   
1                                  /ORGANIZATION/-QOUNTER   
2                                  /organization/-qounter   
3                     /ORGANIZATION/-THE-ONE-OF-THEM-INC-   
4                                   /organization/0-6-com   
5                          /ORGANIZATION/004-TECHNOLOGIES   
6                        /organization/01games-technology   
7                     /ORGANIZATION/0NDINE-BIOMEDICAL-INC   
8                     /organization/0ndine-biomedical-inc   
9                                    /ORGANIZATION/0XDATA   
10                                   /organization/0xdata   
11                                   /ORGANIZATION/0XDATA   
12                                   /organization/0xdata   
13                                        /ORGANIZATION/1   
14                                        /organization/1   
15                                        /ORGANIZATION/1   
16                              /organization/1-2-3-listo   
17                                  /ORGANIZATION/1-4-ALL   
18                         /organization/1-618-technology   
19                            /ORGANIZATION/1-800-DENTIST   
20                            /organization/1-800-doctors   
21               /ORGANIZATION/1-800-PUBLICRELATIONS-INC-   
22                             /organization/1-mainstream   
23                                  /ORGANIZATION/1-OF-99   
24                              /organization/10-20-media   
25                              /ORGANIZATION/10-20-MEDIA   
26                              /organization/10-20-media   
27                              /ORGANIZATION/10-20-MEDIA   
28                          /organization/10-minutes-with   
29                          /ORGANIZATION/10-MINUTES-WITH   
...                                                   ...   
114919                              /ORGANIZATION/ZYNSTRA   
114920                           /organization/zyomyx-inc   
114921                           /ORGANIZATION/ZYOMYX-INC   
114922                           /organization/zyomyx-inc   
114923                           /ORGANIZATION/ZYOMYX-INC   
114924                                 /organization/zype   
114925                                 /ORGANIZATION/ZYPE   
114926                               /organization/zypsee   
114927                               /ORGANIZATION/ZYPSEE   
114928                     /organization/zyraz-technology   
114929                     /ORGANIZATION/ZYRAZ-TECHNOLOGY   
114930                     /organization/zyraz-technology   
114931                     /ORGANIZATION/ZYRAZ-TECHNOLOGY   
114932                                /organization/zyrra   
114933                                /ORGANIZATION/ZYRRA   
114934                                /organization/zyrra   
114935                                /ORGANIZATION/ZYRRA   
114936                               /organization/zystor   
114937                           /ORGANIZATION/ZYTOPROTEC   
114938                                /organization/zzish   
114939                                /ORGANIZATION/ZZISH   
114940                                /organization/zzish   
114941  /ORGANIZATION/ZZNODE-SCIENCE-AND-TECHNOLOGY-CO...   
114942                          /organization/zzzzapp-com   
114943                          /ORGANIZATION/ZZZZAPP-COM   
114944                          /organization/zzzzapp-com   
114945                          /ORGANIZATION/ZZZZAPP-COM   
114946                                 /organization/eron   
114947                               /ORGANIZATION/ASYS-2   
114948  /organization/novatiff-reklam-ve-tantm-hizmetl...   

                                funding_round_permalink   funding_round_type  \
0       /funding-round/9a01d05418af9f794eebff7ace91f638              venture   
1       /funding-round/22dacff496eb7acb2b901dec1dfe5633              venture   
2       /funding-

### How many unique companies are present in rounds2?

In [69]:
rounds2.company_permalink.str.lower().nunique()

66368

*There are 66368 unique companies present in rounds2*

### How many unique companies are present in the companies file?

In [70]:
companies.permalink.str.lower().nunique()

66368

*There are 66368 unique companies present in rounds2*

### Are there any companies in the rounds2 file which are not  present in companies ? Answer Y/N.


In [71]:
#converting the column to lower case for comparison
rounds2['company_permalink'] = rounds2['company_permalink'].str.lower()
companies['permalink'] = companies['permalink'].str.lower()

In [72]:
#number of companies in rounds2 data frame which are not there in companies data frame
len(set(rounds2['company_permalink']) - set(companies['permalink']))

0

*There are no extra companies in rounds2 dataframe compared with companies dataframe .*<br><br>
___Note: Doing difference operations on two sets (A-B) returns only the unique values in A. Hence we can sfely say that all the companies in rounds2 data set are present in companies data set.

### Merge the two data frames so that all  variables (columns)  in the companies frame are added to the rounds2 data frame. Name the merged frame master_frame. How many observations are present in master_frame ?

In [73]:
master_frame = pd.merge(rounds2, companies, how = 'inner', left_on = 'company_permalink', right_on = 'permalink')

In [74]:
master_frame.head()

company_permalink  \
0                  /organization/-fame   
1               /organization/-qounter   
2               /organization/-qounter   
3  /organization/-the-one-of-them-inc-   
4                /organization/0-6-com   

                           funding_round_permalink funding_round_type  \
0  /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1  /funding-round/22dacff496eb7acb2b901dec1dfe5633            venture   
2  /funding-round/b44fbb94153f6cdef13083530bb48030               seed   
3  /funding-round/650b8f704416801069bb178a1418776b            venture   
4  /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   

  funding_round_code   funded_at  raised_amount_usd  \
0                  B  05-01-2015         10000000.0   
1                  A  14-10-2014                NaN   
2                NaN  01-03-2014           700000.0   
3                  B  30-01-2014          3406878.0   
4                  A  19-03-2008          2000000.0   

                             permalink                    name  \
0                  /organization/-fame                   #fame   
1               /organization/-qounter                :Qounter   
2               /organization/-qounter                :Qounter   
3  /organization/-the-one-of-them-inc-  (THE) ONE of THEM,Inc.   
4                /organization/0-6-com                 0-6.com   

             homepage_url                                      category_list  \
0      http://livfame.com                                              Media   
1  http://www.qounter.com  Application Platforms|Real Time|Social Network...   
2  http://www.qounter.com  Application Platforms|Real Time|Social Network...   
3     http://oneofthem.jp                                  Apps|Games|Mobile   
4      http://www.0-6.com                                        Curated Web   

      status country_code state_code      region           city  founded_at  
0  operating          IND         16      Mumbai         Mumbai         NaN  
1  operating          USA         DE  DE - Other  Delaware City  04-09-2014  
2  operating          USA         DE  DE - Other  Delaware City  04-09-2014  
3  operating          NaN        NaN         NaN            NaN         NaN  
4  operating          CHN         22     Beijing        Beijing  01-01-2007

In [75]:
master_frame.shape

(114949, 16)

##### Null value treatment

In [76]:
master_frame.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 114949 entries, 0 to 114948
Data columns (total 16 columns):
company_permalink          114949 non-null object
funding_round_permalink    114949 non-null object
funding_round_type         114949 non-null object
funding_round_code         31140 non-null object
funded_at                  114949 non-null object
raised_amount_usd          94959 non-null float64
permalink                  114949 non-null object
name                       114948 non-null object
homepage_url               108815 non-null object
category_list              111539 non-null object
status                     114949 non-null object
country_code               106271 non-null object
state_code                 104003 non-null object
region                     104782 non-null object
city                       104785 non-null object
founded_at                 94428 non-null object
dtypes: float64(1), object(15)
memory usage: 14.9+ MB


In [77]:
master_frame.isnull().sum(axis = 0)

company_permalink              0
funding_round_permalink        0
funding_round_type             0
funding_round_code         83809
funded_at                      0
raised_amount_usd          19990
permalink                      0
name                           1
homepage_url                6134
category_list               3410
status                         0
country_code                8678
state_code                 10946
region                     10167
city                       10164
founded_at                 20521
dtype: int64

In [78]:
master_frame.isnull().sum(axis = 1)

0         1
1         1
2         1
3         5
4         0
5         2
6         3
7         1
8         1
9         1
10        0
11        1
12        0
13        1
14        2
15        1
16        1
17        4
18        2
19        2
20        1
21        1
22        1
23        2
24        1
25        1
26        1
27        1
28        0
29        1
         ..
114919    1
114920    0
114921    1
114922    1
114923    1
114924    1
114925    1
114926    1
114927    2
114928    1
114929    1
114930    1
114931    0
114932    2
114933    2
114934    2
114935    2
114936    1
114937    0
114938    1
114939    1
114940    1
114941    1
114942    1
114943    1
114944    1
114945    1
114946    6
114947    1
114948    6
Length: 114949, dtype: int64

In [79]:
round(100*master_frame.isnull().sum(axis = 0)/len(master_frame.index))

company_permalink           0.0
funding_round_permalink     0.0
funding_round_type          0.0
funding_round_code         73.0
funded_at                   0.0
raised_amount_usd          17.0
permalink                   0.0
name                        0.0
homepage_url                5.0
category_list               3.0
status                      0.0
country_code                8.0
state_code                 10.0
region                      9.0
city                        9.0
founded_at                 18.0
dtype: float64

In [80]:
#dropping funding_round_code column as not required for analysis
master_frame = master_frame.drop('funding_round_code', axis = 1)
print(master_frame.shape,'\n\n')
print(round(100*master_frame.isnull().sum(axis = 0)/len(master_frame.index)))

(114949, 15) 


company_permalink           0.0
funding_round_permalink     0.0
funding_round_type          0.0
funded_at                   0.0
raised_amount_usd          17.0
permalink                   0.0
name                        0.0
homepage_url                5.0
category_list               3.0
status                      0.0
country_code                8.0
state_code                 10.0
region                      9.0
city                        9.0
founded_at                 18.0
dtype: float64


#### Rows naving null country code

In [81]:
master_frame['country_code'] = master_frame['country_code'].astype('category')
round(100*master_frame['country_code'].value_counts()/len(master_frame.index),2)

USA    62.90
GBR     5.19
CAN     2.71
CHN     2.34
IND     1.94
DEU     1.52
FRA     1.44
ISR     1.42
ESP     1.07
AUS     0.66
IRL     0.63
NLD     0.63
SGP     0.63
BRA     0.61
SWE     0.59
RUS     0.58
JPN     0.56
ITA     0.51
CHE     0.49
KOR     0.47
DNK     0.40
FIN     0.39
CHL     0.39
ARG     0.29
BEL     0.28
HKG     0.27
TUR     0.22
MEX     0.20
NOR     0.19
BGR     0.18
       ...  
KWT     0.00
MAF     0.00
CIV     0.00
ZMB     0.00
KAZ     0.00
MOZ     0.00
BLZ     0.00
NIC     0.00
HND     0.00
PSE     0.00
BAH     0.00
QAT     0.00
GGY     0.00
SEN     0.00
SOM     0.00
SYC     0.00
TGO     0.00
UZB     0.00
PRY     0.00
BRN     0.00
OMN     0.00
BLM     0.00
MNE     0.00
BRB     0.00
DZA     0.00
JEY     0.00
IRN     0.00
DMA     0.00
GRD     0.00
KNA     0.00
Name: country_code, Length: 137, dtype: float64

In [82]:
#imputing the missing values with USA
master_frame.loc[master_frame['country_code'].isnull(),'country_code'] = 'USA'
print(master_frame.shape,'\n\n')
print(round(100*master_frame.isnull().sum(axis = 0)/len(master_frame.index)))


(114949, 15) 


company_permalink           0.0
funding_round_permalink     0.0
funding_round_type          0.0
funded_at                   0.0
raised_amount_usd          17.0
permalink                   0.0
name                        0.0
homepage_url                5.0
category_list               3.0
status                      0.0
country_code                0.0
state_code                 10.0
region                      9.0
city                        9.0
founded_at                 18.0
dtype: float64


#### Rows having null state code

In [83]:
#master_frame['state_code'] = master_frame['state_code'].astype('category')
round(100*master_frame['state_code'].value_counts()/len(master_frame.index),2)

CA    23.20
NY     6.39
MA     5.04
TX     3.19
H9     2.79
WA     2.23
7      1.90
FL     1.75
2      1.69
PA     1.68
IL     1.59
CO     1.59
ON     1.27
GA     1.16
VA     1.13
NC     1.09
5      1.09
MD     1.06
NJ     1.05
OH     1.03
22     1.01
A8     0.95
16     0.95
TN     0.83
MN     0.74
UT     0.73
4      0.69
19     0.68
23     0.67
11     0.67
      ...  
V3     0.00
V9     0.00
V7     0.00
O2     0.00
D7     0.00
75     0.00
L8     0.00
W3     0.00
Q6     0.00
62     0.00
Y7     0.00
W2     0.00
GU     0.00
P1     0.00
PE     0.00
T9     0.00
V6     0.00
U1     0.00
Y5     0.00
L7     0.00
P5     0.00
J5     0.00
O1     0.00
F5     0.00
Z8     0.00
Z3     0.00
O3     0.00
79     0.00
44     0.00
WE     0.00
Name: state_code, Length: 311, dtype: float64

In [84]:
#master_frame.loc[master_frame['city_code'].isnull(),'city_code'] = 'CA'
#print(master_frame.shape,'\n\n')
#print(round(100*master_frame.isnull().sum(axis = 0)/len(master_frame.index)))


#dropping as no state level analysis required
master_frame = master_frame.drop('state_code', axis = 1)
print(master_frame.shape,'\n\n')
print(round(100*master_frame.isnull().sum(axis = 0)/len(master_frame.index)))

(114949, 14) 


company_permalink           0.0
funding_round_permalink     0.0
funding_round_type          0.0
funded_at                   0.0
raised_amount_usd          17.0
permalink                   0.0
name                        0.0
homepage_url                5.0
category_list               3.0
status                      0.0
country_code                0.0
region                      9.0
city                        9.0
founded_at                 18.0
dtype: float64


#### Rows having null region

In [85]:
#master_frame['region'] = master_frame['region'].astype('category')
round(100*master_frame['region'].value_counts()/len(master_frame.index),2)

SF Bay Area            16.60
New York City           5.82
Boston                  4.76
London                  3.48
Los Angeles             2.89
Seattle                 2.16
Washington, D.C.        1.71
San Diego               1.61
Chicago                 1.46
Austin                  1.44
Denver                  1.43
Atlanta                 1.12
Tel Aviv                1.09
Toronto                 1.05
Beijing                 0.96
Dallas                  0.94
Newark                  0.90
Philadelphia            0.89
Anaheim                 0.88
Paris                   0.87
Raleigh                 0.78
Salt Lake City          0.70
Minneapolis             0.66
Singapore               0.60
Pittsburgh              0.59
Portland, Oregon        0.57
Hartford                0.57
Bangalore               0.56
Houston                 0.56
Shanghai                0.55
                       ...  
Orleans                 0.00
Hungerford              0.00
Lviv                    0.00
Oviedo        

In [86]:
#dropping as no region level analysis required
master_frame = master_frame.drop('region', axis = 1)
print(master_frame.shape,'\n\n')
print(round(100*master_frame.isnull().sum(axis = 0)/len(master_frame.index)))

(114949, 13) 


company_permalink           0.0
funding_round_permalink     0.0
funding_round_type          0.0
funded_at                   0.0
raised_amount_usd          17.0
permalink                   0.0
name                        0.0
homepage_url                5.0
category_list               3.0
status                      0.0
country_code                0.0
city                        9.0
founded_at                 18.0
dtype: float64


#### Rows having null city

In [87]:
#master_frame['city'] = master_frame['city'].astype('category')
round(100*master_frame['city'].value_counts()/len(master_frame.index),2)

San Francisco             6.46
New York                  5.34
London                    2.80
Seattle                   1.39
Austin                    1.36
Palo Alto                 1.35
Cambridge                 1.30
Boston                    1.25
Mountain View             1.15
San Diego                 1.15
Los Angeles               1.08
Chicago                   1.05
Beijing                   0.96
San Jose                  0.91
Sunnyvale                 0.83
Paris                     0.77
Santa Clara               0.76
Toronto                   0.75
San Mateo                 0.74
Redwood City              0.72
Atlanta                   0.72
Singapore                 0.59
Shanghai                  0.52
Bangalore                 0.52
Menlo Park                0.51
Pittsburgh                0.51
Dublin                    0.49
Santa Monica              0.48
Tel Aviv                  0.47
Irvine                    0.47
                          ... 
Asti                      0.00
Brenham 

In [88]:
#dropping as no city level analysis required
master_frame = master_frame.drop('city', axis = 1)
print(master_frame.shape,'\n\n')
print(round(100*master_frame.isnull().sum(axis = 0)/len(master_frame.index)))

(114949, 12) 


company_permalink           0.0
funding_round_permalink     0.0
funding_round_type          0.0
funded_at                   0.0
raised_amount_usd          17.0
permalink                   0.0
name                        0.0
homepage_url                5.0
category_list               3.0
status                      0.0
country_code                0.0
founded_at                 18.0
dtype: float64


In [89]:
#dropping founded_at
master_frame = master_frame.drop('founded_at', axis = 1)
print(master_frame.shape,'\n\n')
print(round(100*master_frame.isnull().sum(axis = 0)/len(master_frame.index)))

(114949, 11) 


company_permalink           0.0
funding_round_permalink     0.0
funding_round_type          0.0
funded_at                   0.0
raised_amount_usd          17.0
permalink                   0.0
name                        0.0
homepage_url                5.0
category_list               3.0
status                      0.0
country_code                0.0
dtype: float64


In [90]:
#dropping homepage_url
master_frame = master_frame.drop('homepage_url', axis = 1)
print(master_frame.shape,'\n\n')
print(round(100*master_frame.isnull().sum(axis = 0)/len(master_frame.index)))

(114949, 10) 


company_permalink           0.0
funding_round_permalink     0.0
funding_round_type          0.0
funded_at                   0.0
raised_amount_usd          17.0
permalink                   0.0
name                        0.0
category_list               3.0
status                      0.0
country_code                0.0
dtype: float64


#### Null raised_amount_usd

In [91]:
master_frame['raised_amount_usd'].describe()

count    9.495900e+04
mean     1.042687e+07
std      1.148212e+08
min      0.000000e+00
25%      3.225000e+05
50%      1.680511e+06
75%      7.000000e+06
max      2.127194e+10
Name: raised_amount_usd, dtype: float64

#### Deleting the rows having null raised_amount_usd

In [92]:
master_frame = master_frame.drop(master_frame.loc[master_frame['raised_amount_usd'].isnull()].index, axis = 0)
print(master_frame.shape,'\n\n')
print(round(100*master_frame.isnull().sum(axis = 0)/len(master_frame.index)))

(94959, 10) 


company_permalink          0.0
funding_round_permalink    0.0
funding_round_type         0.0
funded_at                  0.0
raised_amount_usd          0.0
permalink                  0.0
name                       0.0
category_list              1.0
status                     0.0
country_code               0.0
dtype: float64


#### Deleting rows having null category list

In [93]:
master_frame = master_frame.drop(master_frame.loc[master_frame['category_list'].isnull()].index, axis = 0)
print(master_frame.shape,'\n\n')
print(round(100*master_frame.isnull().sum(axis = 0)/len(master_frame.index)))

(93915, 10) 


company_permalink          0.0
funding_round_permalink    0.0
funding_round_type         0.0
funded_at                  0.0
raised_amount_usd          0.0
permalink                  0.0
name                       0.0
category_list              0.0
status                     0.0
country_code               0.0
dtype: float64


In [94]:
#percentage of data left
round(100*len(master_frame.index)/114948,2)

81.7

# Checkpoint 2: Funding Type Analysis

In [95]:
#filtering out the desired funding_round_types
investment_type_filtered = master_frame.loc[master_frame['funding_round_type'].isin(['venture','seed','angel','private_equity']),:]


In [153]:
master_frame.to_csv("C:/Users/vemul/Desktop/DataScience/Investment Case Study/Master.csv", encoding='utf-8',index=False)

In [96]:
#Grouping by funding_round_type and getting the mean in millions, rounded off to 2 decimal places
round((investment_type_filtered.groupby('funding_round_type').raised_amount_usd.mean()/10**6),2).sort_values(ascending = False)

funding_round_type
private_equity    73.39
venture           11.72
angel              0.96
seed               0.72
Name: raised_amount_usd, dtype: float64

# Checkpoint 3: Country Analysis

In [97]:
#list of country codes of english speaking country
english_speaking_cc = ['BWA','CMR','ETH','ERI','GMB','GHA','KEN','LSO','LBR','MWI','MUS','NAM','NGA','RWA','SYC','SLE','ZAF',
'SSD','SDN','SWZ','TZA','UGA','ZMB','ZWE','ATG','BHS','BRB','BLZ','CAN','DMA','GRD','GUY','JAM','KNA','LCA','VCT','TTO','USA',
'IND','PAK','PHL','SGP','AUS','FJI','KIR','MHL','FSM','NRU','NZL','PLW','PNG','WSM','SLB','TON','TUV','VUT','IRL','MLT','GBR']

In [98]:
#adding a new column to master_frame 'english_speaking' whith value 1 for yes and 0 for no
master_frame['english_speaking'] = master_frame.country_code.isin(english_speaking_cc)

In [99]:
#filtering with the requied conditions
english_venture = master_frame.loc[master_frame['funding_round_type'].isin(['venture']) & master_frame['english_speaking'] == True, ['country_code','raised_amount_usd']]
#grouping by country_code, aggregating the sum for grouped countries in millions with 2 decimal places and saving the the top 9 countries
top9 = pd.DataFrame(round(english_venture.groupby('country_code').raised_amount_usd.sum()/10**6,2)).nlargest(9,'raised_amount_usd')
top9

raised_amount_usd
country_code                   
USA                   442187.92
GBR                    20072.81
IND                    14261.51
CAN                     9482.22
SGP                     2793.92
IRL                     1669.29
AUS                     1319.03
NZL                      448.32
ZAF                      233.71

Hence the top three countries based on total investment are USA, Breat Britain and India

# Checkpoint 4: Sector Analysis 1

In [100]:
#splitting the primary sector from the category list.
primary_sector_split = master_frame['category_list'].str.split('|', n = 1, expand = True)
#creating a new column and assigning the primary sector split above
master_frame['primary_sector'] = primary_sector_split[0]

In [105]:
#loading the mapping file
mapping = pd.read_csv("C:/Users/vemul/Desktop/DataScience/Investment Case Study/mapping.csv")

In [106]:
mapping.head()

category_list  Automotive & Sports  Blanks  Cleantech / Semiconductors  \
0            NaN                    0       1                           0   
1             3D                    0       0                           0   
2    3D Printing                    0       0                           0   
3  3D Technology                    0       0                           0   
4     Accounting                    0       0                           0   

   Entertainment  Health  Manufacturing  News, Search and Messaging  Others  \
0              0       0              0                           0       0   
1              0       0              1                           0       0   
2              0       0              1                           0       0   
3              0       0              1                           0       0   
4              0       0              0                           0       0   

   Social, Finance, Analytics, Advertising  
0                                        0  
1                                        0  
2                                        0  
3                                        0  
4                                        1

In [107]:
#on analysing the category list in mapping data set it found that 0 is there in words instead of 'na'.
#so replaceing '0' with 'na'
mapping['category_list'] = mapping['category_list'].str.replace('0','na')

In [108]:
#menting data frame from wide to long form
mapping_melted = pd.melt(mapping, id_vars = ['category_list'], var_name= 'main_sector').sort_values(by = 'value', ascending = False)

In [109]:
mapping_melted

category_list                              main_sector  \
3096                     Hospitals                                   Health   
5452                       Systems                                   Others   
5382                Rental Housing                                   Others   
5379                      Religion                                   Others   
2588                   Photography                            Entertainment   
5376                    Recreation                                   Others   
5375                      Realtors                                   Others   
5374                     Real Time                                   Others   
5373                   Real Estate                                   Others   
5372                      QR Codes                                   Others   
2587                 Photo Sharing                            Entertainment   
2586                 Photo Editing                            Entertainment   
5364            Project Management                                   Others   
5363         Professional Services                                   Others   
5361         Productivity Software                                   Others   
5360                  Productivity                                   Others   
5359  Product Development Services                                   Others   
5358                Product Design                                   Others   
5357                   Procurement                                   Others   
5355                Private School                                   Others   
5354                       Privacy                                   Others   
5353                      Printing                                   Others   
5351                 Presentations                                   Others   
3757                     Groceries                            Manufacturing   
5348   Postal and Courier Services                                   Others   
5347                       Portals                                   Others   
5346                       Polling                                   Others   
5345                      Politics                                   Others   
5344                 Point of Sale                                   Others   
5343                      Plumbers                                   Others   
...                            ...                                      ...   
2044               Video on Demand               Cleantech / Semiconductors   
2043                   Video Games               Cleantech / Semiconductors   
2042            Video Conferencing               Cleantech / Semiconductors   
2041                         Video               Cleantech / Semiconductors   
2040                    Veterinary               Cleantech / Semiconductors   
2039               Venture Capital               Cleantech / Semiconductors   
2062                         Women               Cleantech / Semiconductors   
2064                           NaN                            Entertainment   
2085                        Alumni                            Entertainment   
2065                            3D                            Entertainment   
2084          Alternative Medicine                            Entertainment   
2083                  All Students                            Entertainment   
2082                   All Markets                            Entertainment   
2081                    Algorithms                            Entertainment   
2080         Air Pollution Control                            Entertainment   
2079                   Agriculture                            Entertainment   
2078                     Aerospace                            Entertainment   
2077                        Advice                            Entertainment   
2076         Advertising Platforms                            Entertain

In [110]:
#selecting only those roes with values = 1
mapping_melted_1 = mapping_melted.loc[mapping_melted['value']==1,['category_list','main_sector']]

In [111]:
#renaming column category_list to primary_sector
mapping_melted_1 = mapping_melted_1.rename(columns = {'category_list':'primary_sector'})

In [112]:
mapping_melted_1

primary_sector                              main_sector
3096                     Hospitals                                   Health
5452                       Systems                                   Others
5382                Rental Housing                                   Others
5379                      Religion                                   Others
2588                   Photography                            Entertainment
5376                    Recreation                                   Others
5375                      Realtors                                   Others
5374                     Real Time                                   Others
5373                   Real Estate                                   Others
5372                      QR Codes                                   Others
2587                 Photo Sharing                            Entertainment
2586                 Photo Editing                            Entertainment
5364            Project Management                                   Others
5363         Professional Services                                   Others
5361         Productivity Software                                   Others
5360                  Productivity                                   Others
5359  Product Development Services                                   Others
5358                Product Design                                   Others
5357                   Procurement                                   Others
5355                Private School                                   Others
5354                       Privacy                                   Others
5353                      Printing                                   Others
5351                 Presentations                                   Others
3757                     Groceries                            Manufacturing
5348   Postal and Courier Services                                   Others
5347                       Portals                                   Others
5346                       Polling                                   Others
5345                      Politics                                   Others
5344                 Point of Sale                                   Others
5343                      Plumbers                                   Others
...                            ...                                      ...
3188               Medical Devices                                   Health
2792               Assisted Living                                   Health
6017         Performance Marketing  Social, Finance, Analytics, Advertising
5736             Email Newsletters  Social, Finance, Analytics, Advertising
5735               Email Marketing  Social, Finance, Analytics, Advertising
189                      Designers                      Automotive & Sports
188                         Design                      Automotive & Sports
3544               Civil Engineers                            Manufacturing
5739    Employer Benefits Programs  Social, Finance, Analytics, Advertising
6015                      Payments  Social, Finance, Analytics, Advertising
6018             Personal Branding  Social, Finance, Analytics, Advertising
6038          Predictive Analytics  Social, Finance, Analytics, Advertising
3187                       Medical                                   Health
3056               Genetic Testing                                   Health
6020              Personal Finance  Social, Finance, Analytics, Advertising
6012                          PaaS  Social, Finance, Analytics, Advertising
6037                      Pre Seed  Social, Finance, Analytics, Advertising
14                       Aerospace                      Automotive & Sports
6011            P2P Money Transfer  Social, Finance, Analytics, Advertising
5704            Direct Advertising  Social, Finance, Analytics, Advertising
3338                 Senior Health                                  

In [114]:
#removing the main_sector 'Blanks'
mapping_melted_1 = mapping_melted_1.drop(mapping_melted_1.loc[mapping_melted_1['main_sector']=='Blanks'].index,axis =0)

In [115]:
master_frame.shape

(93915, 12)

In [116]:
#merging
master_frame = pd.merge(master_frame, mapping_melted_1, how = 'inner', on = 'primary_sector')

In [117]:
master_frame.head()

company_permalink  \
0           /organization/-fame   
1           /organization/90min   
2           /organization/90min   
3           /organization/90min   
4  /organization/a-dance-for-me   

                           funding_round_permalink   funding_round_type  \
0  /funding-round/9a01d05418af9f794eebff7ace91f638              venture   
1  /funding-round/21a2cbf6f2fb2a1c2a61e04bf930dfe6              venture   
2  /funding-round/bd626ed022f5c66574b1afe234f3c90d              venture   
3  /funding-round/fd4b15e8c97ee2ffc0acccdbe1a98810              venture   
4  /funding-round/9ab9dbd17bf010c79d8415b2c22be6fa  equity_crowdfunding   

    funded_at  raised_amount_usd                     permalink  \
0  05-01-2015         10000000.0           /organization/-fame   
1  06-10-2015         15000000.0           /organization/90min   
2  07-05-2013          5800000.0           /organization/90min   
3  26-03-2014         18000000.0           /organization/90min   
4  26-03-2014          1090000.0  /organization/a-dance-for-me   

             name                        category_list     status  \
0           #fame                                Media  operating   
1           90min  Media|News|Publishing|Soccer|Sports  operating   
2           90min  Media|News|Publishing|Soccer|Sports  operating   
3           90min  Media|News|Publishing|Soccer|Sports  operating   
4  A Dance for Me       Media|News|Photo Sharing|Video  operating   

  country_code  english_speaking primary_sector    main_sector  
0          IND              True          Media  Entertainment  
1          GBR              True          Media  Entertainment  
2          GBR              True          Media  Entertainment  
3          GBR              True          Media  Entertainment  
4          USA              True          Media  Entertainment

In [118]:
master_frame.shape

(93571, 13)

# Checkpoint 5: Sector Analysis 2

In [119]:
#C1 = 'USA' for FT = 'venture'
D1 = master_frame.loc[master_frame['country_code'].isin(['USA']) & master_frame['funding_round_type'].isin(['venture']), :]
#C2 = 'GBR' for FT = 'venture'
D2 = master_frame.loc[master_frame['country_code'].isin(['GBR']) & master_frame['funding_round_type'].isin(['venture']), :]
#C3 = 'IND' for FT = 'venture'
D3 = master_frame.loc[master_frame['country_code'].isin(['IND']) & master_frame['funding_round_type'].isin(['venture']), :]

In [121]:
#The total number of investments for each main sector in a separate column
D1 = D1.assign(count_of_investments_usa = D1.groupby('main_sector')['raised_amount_usd'].transform('count'))

#The total amount invested in each main sector in a separate column
D1 = D1.assign(amount_of_investments_usa = round(D1.groupby('main_sector')['raised_amount_usd'].transform('sum')/10**6,2))

In [122]:
D1.head()

company_permalink  \
7   /organization/all-def-digital   
11      /organization/audiosocket   
12      /organization/audiosocket   
13      /organization/audiosocket   
16           /organization/babble   

                            funding_round_permalink funding_round_type  \
7   /funding-round/452a2342fe720285c3b92e9bd927d9ba            venture   
11  /funding-round/0959f59e3a89bc8db8c6ed9c646349fd            venture   
12  /funding-round/4e3a53eec6c609943df752206070b23f            venture   
13  /funding-round/bc0b412cf2fa7b739e4105ad756b47a4            venture   
16  /funding-round/43cc678b39be3e684bb88501af05ff9b            venture   

     funded_at  raised_amount_usd                      permalink  \
7   06-08-2014          5000000.0  /organization/all-def-digital   
11  02-05-2013           650000.0      /organization/audiosocket   
12  02-09-2009           750000.0      /organization/audiosocket   
13  15-02-2012          1582300.0      /organization/audiosocket   
16  05-02-2009          2000000.0           /organization/babble   

               name           category_list     status country_code  \
7   All Def Digital                   Media  operating          USA   
11      Audiosocket  Media|Music|Technology  operating          USA   
12      Audiosocket  Media|Music|Technology  operating          USA   
13      Audiosocket  Media|Music|Technology  operating          USA   
16           Babble    Media|News|Parenting   acquired          USA   

    english_speaking primary_sector    main_sector  count_of_investments_usa  \
7               True          Media  Entertainment                      1950   
11              True          Media  Entertainment                      1950   
12              True          Media  Entertainment                      1950   
13              True          Media  Entertainment                      1950   
16              True          Media  Entertainment                      1950   

    amount_of_investments_usa  
7                    18231.26  
11                   18231.26  
12                   18231.26  
13                   18231.26  
16                   18231.26

In [123]:
#The total number of investments for each main sector in a separate column
D2 = D2.assign(count_of_investments_gbr = D2.groupby('main_sector')['raised_amount_usd'].transform('count'))

#The total amount invested in each main sector in a separate column
D2 = D2.assign(amount_of_investments_gbr = round(D2.groupby('main_sector')['raised_amount_usd'].transform('sum')/10**6,2))

In [124]:
#The total number of investments for each main sector in a separate column
D3 = D3.assign(count_of_investments_ind = D3.groupby('main_sector')['raised_amount_usd'].transform('count'))

#The total amount invested in each main sector in a separate column
D3 = D3.assign(amount_of_investments_ind = round(D3.groupby('main_sector')['raised_amount_usd'].transform('sum')/10**6,2))

In [125]:
D = D1.append(D2.append(D3))

C:\Users\vemul\Anaconda3\lib\site-packages\pandas\core\frame.py:6201: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version
of pandas will change to not sort by default.

To accept the future behavior, pass 'sort=True'.

To retain the current behavior and silence the warning, pass sort=False

  sort=sort)


In [152]:
D.to_csv("C:/Users/vemul/Desktop/DataScience/Investment Case Study/Companies.csv", encoding='utf-8',index=False)

In [132]:
#Total number of Investments (count) for C1
D1['raised_amount_usd'].count()

37697

In [133]:
#Total number of Investments (count) for C2
D2['raised_amount_usd'].count()

2035

In [134]:
#Total number of Investments (count) for C3
D3['raised_amount_usd'].count()

819

In [135]:
#Total amount of investment (USD) for C1
D1['raised_amount_usd'].sum()

440677062249.0

In [136]:
#Total amount of investment (USD) for C2
D2['raised_amount_usd'].sum()

19983803406.0

In [137]:
#Total amount of investment (USD) for C3
D3['raised_amount_usd'].sum()

14261508718.0

##### Pivot 1: Mean investment amount across sectors

In [138]:
round(D.pivot_table(values = 'raised_amount_usd',
             index = 'main_sector',
             columns = 'country_code',
             aggfunc = 'mean')/10**6,2)

country_code                               GBR    IND    USA
main_sector                                                 
Automotive & Sports                      10.93  50.20  21.99
Cleantech / Semiconductors               11.56  14.56  15.09
Entertainment                             8.55  12.84   9.35
Health                                    8.87  13.44   9.76
Manufacturing                             8.35  13.28  10.31
News, Search and Messaging               12.27  13.38  10.65
Others                                    8.71  22.29   9.89
Social, Finance, Analytics, Advertising   8.96  10.74  11.96

##### Pivot 2: Count of investment across sectors

In [139]:
D.pivot_table(values = 'raised_amount_usd',
             index = 'main_sector',
             columns = 'country_code',
             aggfunc = 'count')

country_code                             GBR  IND   USA
main_sector                                            
Automotive & Sports                       44   31   554
Cleantech / Semiconductors               437   54  8128
Entertainment                            136   76  1950
Health                                   118   42  3410
Manufacturing                            122   55  2611
News, Search and Messaging               241  130  4679
Others                                   516  284  8761
Social, Finance, Analytics, Advertising  421  147  7604

In [140]:
#name and number of investments in top 3 sectors in C1
D1.groupby('main_sector')['raised_amount_usd'].count().sort_values(ascending = False).head(5)

main_sector
Others                                     8761
Cleantech / Semiconductors                 8128
Social, Finance, Analytics, Advertising    7604
News, Search and Messaging                 4679
Health                                     3410
Name: raised_amount_usd, dtype: int64

___Note: From pivot 1 we can see that the mean investment amount in sector Cleantech/ Semiconductors for funding rounf type venture is more than 15 million USD. So, the second sector will be Social, Finance, Analytics, Advertising and third will be News, Search and Messaging.___

In [141]:
#name and number of investments in top 3 sectors in C2
D2.groupby('main_sector')['raised_amount_usd'].count().sort_values(ascending = False).head(5)

main_sector
Others                                     516
Cleantech / Semiconductors                 437
Social, Finance, Analytics, Advertising    421
News, Search and Messaging                 241
Entertainment                              136
Name: raised_amount_usd, dtype: int64

In [142]:
#name and number of investments in top 3 sectors in C3
D3.groupby('main_sector')['raised_amount_usd'].count().sort_values(ascending = False).head(5)

main_sector
Others                                     284
Social, Finance, Analytics, Advertising    147
News, Search and Messaging                 130
Entertainment                               76
Manufacturing                               55
Name: raised_amount_usd, dtype: int64

___Note: From pivot 1 we can see that the mean investment amount in sector Others for funding round type venture is more than 15 million USD. So, the first sector will be Social, Finance, Analytics, Advertising, second will be News, Search and Messaging and third will be entertainment.___

In [143]:
#For point 3 (top sector count-wise), which company received the highest investment? --> C1
D1.loc[D1['main_sector']=='Others',:].groupby('name').raised_amount_usd.count().sort_values(ascending = False).head(5)

name
Terascala      13
VectorMAX      12
Lecere         11
BTI Systems    10
Approva         9
Name: raised_amount_usd, dtype: int64

In [144]:
#For point 3 (top sector count-wise), which company received the highest investment? --> C2
D2.loc[D2['main_sector']=='Others',:].groupby('name').raised_amount_usd.count().sort_values(ascending = False).head(5)

name
Rostima           8
Electric Cloud    7
picoChip          6
Helveta           6
SenSage           5
Name: raised_amount_usd, dtype: int64

In [145]:
#For point 3 (top sector count-wise), which company received the highest investment? --> C3
D3.loc[D3['main_sector']=='Social, Finance, Analytics, Advertising',:].groupby('name').raised_amount_usd.count().sort_values(ascending = False).head(5)

name
Manthan Systems    6
Komli Media        4
ShopClues.com      4
PolicyBazaar       4
yepme.com          3
Name: raised_amount_usd, dtype: int64

In [146]:
#For point 4 (second best sector count-wise), which company received the highest investment? --> C1
D1.loc[D1['main_sector']=='Social, Finance, Analytics, Advertising',:].groupby('name').raised_amount_usd.count().sort_values(ascending = False).head(5)

name
edo Interactive    11
XOS Digital        10
InstaMed           10
ChaCha              9
RockYou             9
Name: raised_amount_usd, dtype: int64

In [147]:
#For point 4 (second best sector count-wise), which company received the highest investment? --> C2
D2.loc[D2['main_sector']=='Cleantech / Semiconductors',:].groupby('name').raised_amount_usd.count().sort_values(ascending = False).head(5)

name
Green Biologics                     6
Population Genetics Technologies    6
Horizon Discovery                   5
Sirigen                             5
F-star Biotechnology Limited        5
Name: raised_amount_usd, dtype: int64

In [148]:
#For point 4 (second best sector count-wise), which company received the highest investment? --> C3
D3.loc[D3['main_sector']=='News, Search and Messaging',:].groupby('name').raised_amount_usd.count().sort_values(ascending = False).head(5)

name
Quikr                 6
TinyOwl Technology    4
hike                  3
GupShup               3
Ezetap                3
Name: raised_amount_usd, dtype: int64